src/ripple/app/tx/impl/PayChan.cpp

```diff
@@ -279,37 +312,26 @@ PayChanFund::doApply()
     if (!slep)
         return tecNO_ENTRY;
 
-    AccountID const src = (*slep)[sfAccount];
     auto const txAccount = ctx_.tx[sfAccount];
     auto const expiration = (*slep)[~sfExpiration];
 
+    if (expiration)
     {
-        auto const cancelAfter = (*slep)[~sfCancelAfter];
         auto const closeTime =
             ctx_.view ().info ().parentCloseTime.time_since_epoch ().count ();
-        if ((cancelAfter && closeTime >= *cancelAfter) ||
-            (expiration && closeTime >= *expiration))
+        if (closeTime >= *expiration)
             return closeChannel (
                 slep, ctx_.view (), k.key, ctx_.app.journal ("View"));
     }
 
-    if (src != txAccount)
-        // only the owner can add funds or extend
-        return tecNO_PERMISSION;
+    auto chanMembers = slep->getFieldArray (sfChannelMembers);
 
-    if (auto extend = ctx_.tx[~sfExpiration])
-    {
-        auto minExpiration =
-            ctx_.view ().info ().parentCloseTime.time_since_epoch ().count () +
-                (*slep)[sfSettleDelay];
-        if (expiration && *expiration < minExpiration)
-            minExpiration = *expiration;
+    auto const i = (chanMembers[0].getAccountID (sfAccount) == txAccount)
+        ? 0 : 1;
 
-        if (*extend < minExpiration)
-            return temBAD_EXPIRATION;
-        (*slep)[~sfExpiration] = *extend;
-        ctx_.view ().update (slep);
-    }
+    // only channel members can add funds
+    if (i == 1 && chanMembers[1].getAccountID (sfAccount) != txAccount)
```

Comment:
This code, which does something very obvious, is confusing. It should either be simplified or the comment should better explain what's happening.